packages/web3-eth-accounts/src/rlp/utils.ts

```diff
@@ -0,0 +1,110 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { isHexPrefixed } from 'web3-validator';
+import { stripHexPrefix } from '../common/utils';
+// Global symbols in both browsers and Node.js since v11
+// See https://github.com/microsoft/TypeScript/issues/31535
+declare const TextEncoder: any;
+declare const TextDecoder: any;
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type Input = string | number | bigint | Uint8Array | Array<Input> | null | undefined;
+/** Transform an integer into its hexadecimal value */
+
+const cachedHexes = Array.from({ length: 256 }, (_v, i) => i.toString(16).padStart(2, '0'));
+export const bytesToHex = (uint8a: Uint8Array): string => {
+	// Pre-caching chars with `cachedHexes` speeds this up 6x
+	let hex = '';
+	// eslint-disable-next-line @typescript-eslint/prefer-for-of
+	for (let i = 0; i < uint8a.length; i += 1) {
+		hex += cachedHexes[uint8a[i]];
+	}
+	return hex;
+};
+export const numberToHex = (integer: number | bigint): string => {
+	if (integer < 0) {
+		throw new Error('Invalid integer as argument, must be unsigned!');
+	}
+	const hex = integer.toString(16);
+	return hex.length % 2 ? `0${hex}` : hex;
+};
+export const parseHexByte = (hexByte: string): number => {
+	const byte = Number.parseInt(hexByte, 16);
+	if (Number.isNaN(byte)) throw new Error('Invalid byte sequence');
+	return byte;
+};
+export const hexToBytes = (hex: string): Uint8Array => {
```

Comment:
[hexToBytes](https://github.com/web3/web3.js/blob/de5617ce4f0f50785e075c7638deeac1d6ed615d/packages/web3-utils/src/converters.ts#L140)

Reply:
the same out hexToBytes has validation and fails by error:
`value "f889" at "/0" must pass "bytes" validation`